procal/procal.py

```diff
@@ -477,10 +477,12 @@ def _callback(self, value):
     def _on_changed(self):
 
         try:
-            res = float(eval(self.text()))
-            self._callback(res)
-
-        except:
+            res = eval(self.text())
+            if (isinstance(res, int) or isinstance(res, float)):
+                self._callback(res)
+            else:
+                raise TypeError("Input must evaluate to a number") 
+        except Exception as e:
             # treat all exceptions as syntax error
```

Comment:
Added "Exception as e". Before an exception here caused Euler's number to be printed

Reply:
Hahah good catch